examples/cuda/explicit_cuda_stream.cu

```diff
@@ -0,0 +1,50 @@
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h> // For thrust::device
+#include <thrust/reduce.h>
+#include <thrust/sequence.h>
+
+#include <cuda_runtime.h>
+
+#include <iostream>
+
+// This example shows how to execute a Thrust device algorithm on an explicit
+// CUDA stream. The simple program below fills a vector with the numbers
+// [0, 1000) (thrust::sequence) and then sums them (thrust::reduce), executing
+// both algorithms on the same custom CUDA stream.
+
+int main()
+{
+  thrust::device_vector<int> d_vec(1000);
+
+  // Create the stream:
+  cudaStream_t custom_stream;
+  cudaError_t err = cudaStreamCreate(&custom_stream);
+  if (err != cudaSuccess)
+  {
+    std::cerr << "Error creating stream: " << cudaGetErrorString(err) << "\n";
+    return 1;
+  }
+
+  // Create a new execution policy with the custom stream:
+  auto exec_policy = thrust::device.on(custom_stream);
+
+  // Fill the vector with sequential data.
+  // This will execute using the custom stream.
+  thrust::sequence(exec_policy, d_vec.begin(), d_vec.end());
```

Comment:
```suggestion
  // Construct a new `nosync` execution policy with the custom stream
  auto nosync_exec_policy = thrust::cuda::par_nosync.on(custom_stream);

  // Fill the vector with sequential data.
  // This will execute using the custom stream and the stream will *not* be synchronized before the
  // function returns, meaning asynchronous work may still be executing after returning and the contents of
  // `d_vec` are undefined. 
  thrust::sequence(nosync_exec_policy, d_vec.begin(), d_vec.end());
```